### ISSUE TITLE
How to use input of type datetime-local with Formik's FieldArray

### ISSUE THREAD

**Issue by Jokinen** [jaredpalmer/formik#1318]
## ❓Question

I came across this problem on Chrome. I suspect that it is mainly due to how the `datetime-local` input has its own state.

If you use this sandbox https://codesandbox.io/s/9jp8r30ozy

And repeat this steps, you should be able to reproduce the behaviour

1. Add two new fields.
2. In the first field, with controls under the headline of `Type datetime-local`, press the down chevron to launch Chrome's date picker.
3. Choose any date, but don't choose a time.
4. Click `Remove friend` button for the first field.

What happens: the date you selected is left in view
What should happen: there should be no values in the view

If you look at the state, or the second list containing fields of type text, you can see that Formik's state is correct. In fact, no dates have yet visited it. It's just that the value the input shows does not reflect the content in Formik's state, but its own internal state. React probably ends up using the existing element with existing values due to how the list `index` value is used as a key.

What's a good "formiky" way to avoid this behaviour?

**Comment by jaredpalmer** [comment]
This seems to me like a React bug and doesn't have much to do with Formik, since this would happen with any list and any `datetime-local` input, right?

**Comment by Jokinen** [comment]
I would think so. I wouldn’t consider it to be a bug in Formik or anything of that nature. I tried a quick google with react + datetime-local before posting, but only skimmed the results. 

What maybe makes it a bit more interesting from Formik’s perspective is that with Formik it’s somewhat more acceptable or common to use list index as the key. As a result, it’s less obvious what’s a good way to provide a stable unique key when the index sort-of isn’t specific enough. 

For instance in my case I just generated a random ID—which is not perfect. There could be some handlers which could be used to push stuff into Formik’s state prematurely as well, but I didn’t manage to find anything which worked. 

Reading the reply though, it seems that a good place to start looking is key generation strategies for React in general 👍 I can try and find some datetime-local and React interoperability resources and link them back for future googlers.
